newIDE/app/src/UI/Slideshow/Slideshow.js

```diff
@@ -0,0 +1,215 @@
+  type WidthType,
+} from '../../UI/Reponsive/ResponsiveWindowMeasurer';
+import Measure from 'react-measure';
+import { createStyles, makeStyles } from '@material-ui/core';
+import SlideshowArrow from './SlideshowArrow';
+
+const styles = {
+  itemSkeleton: { borderRadius: 6 },
+};
+
+const getItemLineHeight = ({
+  width,
+  containerWidth,
+  itemDesktopRatio,
+  itemMobileRatio,
+}: {
+  width: WidthType,
+  containerWidth: ?number,
+  itemDesktopRatio: number,
+  itemMobileRatio: number,
+}) => {
+  const defaultContainerWidth = width === 'small' ? 320 : 1024;
+  const containerWidthValue = containerWidth || defaultContainerWidth;
+  const lineHeight =
+    width === 'small'
+      ? containerWidthValue / itemMobileRatio
+      : containerWidthValue / itemDesktopRatio;
+
+  return lineHeight;
+};
+
+const useStylesForImage = () =>
+  makeStyles(theme =>
+    createStyles({
+      root: {
+        '&:hover': {
+          transform: 'scale(1.02)',
+        },
+        '&:focus': {
+          transform: 'scale(1.02)',
+          outline: 'none',
+        },
+      },
+    })
+  )();
+
+type SlideshowProps = {|
+  items: ?Array<{|
+    id: string,
+    imageUrl: string,
+    mobileImageUrl: string,
+    onClick: ?() => void,
+  |}>,
+  itemDesktopRatio: number,
+  itemMobileRatio: number,
+|};
+
+const Slideshow = ({
+  items,
+  itemDesktopRatio,
+  itemMobileRatio,
+}: SlideshowProps) => {
+  const windowWidth = useResponsiveWindowWidth();
+  const [containerWidth, setContainerWidth] = React.useState<?number>(null);
+  const itemLineHeight = getItemLineHeight({
+    width: windowWidth,
+    containerWidth,
+    itemDesktopRatio,
+    itemMobileRatio,
+  });
+  const classesForImage = useStylesForImage();
+
+  const [currentSlide, setCurrentSlide] = React.useState(0);
+
+  const handleLeftArrowClick = React.useCallback(
+    () => {
+      if (!items || items.length === 1) return;
+
+      setCurrentSlide(currentSlide === 0 ? items.length - 1 : currentSlide - 1);
+    },
+    [currentSlide, items]
+  );
+
+  const handleRightArrowClick = React.useCallback(
+    () => {
+      if (!items || items.length === 1) return;
+
+      setCurrentSlide(currentSlide === items.length - 1 ? 0 : currentSlide + 1);
+    },
+    [currentSlide, items]
+  );
+
+  React.useEffect(
+    () => {
+      const interval = setInterval(() => {
+        handleRightArrowClick();
+      }, 5000);
+      return () => clearInterval(interval);
+    },
+    [handleRightArrowClick]
+  );
+
+  if (!items) {
+    // If they're loading, display a skeleton so that it doesn't jump when loaded.
+    return (
+      <Measure
+        bounds
+        onResize={contentRect => {
+          setContainerWidth(contentRect.bounds.width);
+        }}
+      >
+        {({ contentRect, measureRef }) => (
+          <Paper square background="dark">
+            <div
+              ref={measureRef}
+              style={{
+                display: 'flex',
+                flex: 1,
```

Comment:
Should you put most of the styles in the canonical `styles` object?

Reply:
oh yes, forgot to refactor this